include/seqan3/argument_parser/validators.hpp

```diff
@@ -375,13 +374,16 @@ class file_validator_base
         auto drop_less_ext = tmp_str | views::drop(1);
 
         // Compares the extensions in lower case.
-        auto cmp_lambda = [&] (std::string const & cmp)
+        auto case_insensitive_cmp = [&] (std::string const & ext)
         {
-            return std::ranges::equal(cmp | views::to_lower, drop_less_ext | views::to_lower);
+            return std::ranges::equal(ext, drop_less_ext, [](auto && chr1, auto && chr2)
```

Comment:
We expect these to be characters since we are comparing strings right?
then I think we should capture by value not by reference.

:nail_care: If you are being more expressive with the name, you might also want to change it to case_insensitive`_comparator`? :)

:nail_care: space. 
```suggestion
            return std::ranges::equal(ext, drop_less_ext, [] (auto && chr1, auto && chr2)
```